reactor-netty-core/src/main/java/reactor/netty/channel/ChannelMetricsRecorder.java

```diff
@@ -76,4 +76,14 @@
 	 * @param status the status of the operation
 	 */
 	void recordResolveAddressTime(SocketAddress remoteAddress, Duration time, String status);
+
+	/**
+	 * Updates the number of the channels currently opened and accepted by the server
+	 *
+	 * @param localAddress the server local address
+	 * @param amount The amount of server connection to add in the meter
+	 * @since 1.0.15
+	 */
+	default void incrementServerConnections(SocketAddress localAddress, int amount) { }
```

Comment:
@simonbasle What do you think about the API? I'm still thinking that this is a bit ... yeah I don't know.

Increment
`recorder().incrementServerConnections(ctx.channel().localAddress(), -1);`

Decrement
`recorder().incrementServerConnections(ctx.channel().localAddress(), -1);`

Reply:
I guess a single method called increment which accepts a negative number is a bit weird 🤔 

in fact, how did I miss this: `Counter` in micrometer **doesn't support decrementing**:

> Counters report a single metric, a count. The Counter interface allows you to increment by a fixed amount, which must be positive.

Reply:
indeed, maybe 2 methods would make sense (especially given that amount is either -1 or 1). So I suggest:

```java
recordConnectionOpened(SocketAddress localAddress);
recordConnectionClosed(SocketAddress localAddress);
```

see also the Http recorder below